// Copyright (c) 2020 The Orbit Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package orbit_client_protos;

message TimerInfo {
  // NextID: 18
  uint64 start = 1;
  uint64 end = 2;
  uint32 process_id = 3;
  uint32 thread_id = 4;
  uint32 depth = 5;

  enum Type {
    reserved 2, 8, 9, 10;
    kNone = 0;
    kCoreActivity = 1;
    kGpuActivity = 3;
    kFrame = 4;
    kGpuCommandBuffer = 5;
    kGpuDebugMarker = 6;
    kApiEvent = 7;
    kApiScope = 11;
    kApiScopeAsync = 12;
  }
  Type type = 6;

  int32 processor = 7;
  uint64 callstack_id = 8;
  uint64 function_id = 9;
  uint64 user_data_key = 10;
  uint64 timeline_hash = 11;
  repeated uint64 registers = 12;
  Color color = 13;
  uint64 group_id = 14;
  uint64 api_async_scope_id = 15;
  uint64 address_in_function = 16;
  string api_scope_name = 17;
}

message Color {
  // Each color must be between 0 and 255 (including).
  uint32 red = 1;
  uint32 green = 2;
  uint32 blue = 3;
  uint32 alpha = 4;
}
